src/uu/cp/src/cp.rs

```diff
@@ -157,18 +156,49 @@ pub enum CopyMode {
     AttrOnly,
 }
 
-// The ordering here determines the order in which attributes are (re-)applied.
-// In particular, Ownership must be changed first to avoid interfering with mode change.
-#[derive(Clone, Eq, PartialEq, Debug, PartialOrd, Ord)]
-pub enum Attribute {
+#[derive(Debug)]
+pub struct Attributes {
     #[cfg(unix)]
-    Ownership,
-    Mode,
-    Timestamps,
-    #[cfg(feature = "feat_selinux")]
-    Context,
-    Links,
-    Xattr,
+    ownership: Preserve,
+    mode: Preserve,
+    timestamps: Preserve,
+    context: Preserve,
+    links: Preserve,
+    xattr: Preserve,
+}
+
+impl Attributes {
+    pub(crate) fn max(&mut self, other: Self) {
+        #[cfg(unix)]
+        {
+            self.ownership = self.ownership.max(other.ownership);
+        }
+        self.mode = self.mode.max(other.mode);
+        self.timestamps = self.timestamps.max(other.timestamps);
+        self.context = self.context.max(other.context);
+        self.links = self.links.max(other.links);
+        self.xattr = self.xattr.max(other.xattr);
+    }
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum Preserve {
+    No,
+    Yes { required: bool },
+}
+
+impl Preserve {
```

Comment:
could you please add a comment here to explain what it is doing? thanks

Reply:
Done